Client/SafeBrowsing/GSafeBrowsing.swift

```diff
@@ -0,0 +1,366 @@
+    private static let version = AppInfo.appVersion
+    
+    private let baseURL = "https://safebrowsing.brave.com"
+    private let session = URLSession(configuration: .ephemeral)
+    private let database = SafeBrowsingDatabase()
+    
+    public static let shared = SafeBrowsingClient()
+    
+    private init() {
+        self.database.scheduleUpdate { [weak self] in
+            self?.fetch({
+                if let error = $0 {
+                    log.error(error)
+                }
+            })
+        }
+    }
+    
+    func find(_ hashes: [String], _ completion: @escaping (_ isSafe: Bool, Error?) -> Void) {
+        database.find(hashes, completion: { [weak self] discoveredHashes in
+            guard let self = self else { return completion(true, nil) }
+            
+            if discoveredHashes.isEmpty {
+                return DispatchQueue.global(qos: .background).async {
+                    completion(true, nil)
+                }
+            }
+            
+            if !self.database.canFind() {
+                return DispatchQueue.global(qos: .background).async {
+                    completion(false, nil)
+                }
+            }
+            
+            let clientInfo = ClientInfo(clientId: SafeBrowsingClient.clientId,
+                                        clientVersion: SafeBrowsingClient.version)
+            
+            let threatTypes: [ThreatType] = [.malware,
+                                             .socialEngineering,
+                                             .unwantedSoftware,
+                                             .potentiallyHarmfulApplication]
+            
+            let platformTypes: [PlatformType] = [.any, .ios]
+            let threatEntryTypes: [ThreatEntryType] = [.url, .exe]
+            
+            let threatInfo = ThreatInfo(threatTypes: threatTypes,
+                                        platformTypes: platformTypes,
+                                        threatEntryTypes: threatEntryTypes,
+                                        threatEntries: discoveredHashes.map {
+                                            return ThreatEntry(hash: $0, url: nil, digest: nil)
+                }
+            )
+            
+            do {
+                let body = FindRequest(client: clientInfo, threatInfo: threatInfo)
+                let request = try self.encode(.post, endpoint: .fullHashes, body: body)
+                self.executeRequest(request, type: FindResponse.self) { [weak self] response, error in
+                    guard let self = self else { return }
+                    
+                    if error != nil {
+                        self.database.enterBackoffMode(.find)
+                    }
+                    
+                    DispatchQueue.global(qos: .background).async {
+                        if let error = error {
+                            return completion(false, error)
```

Comment:
Similarly here, if there's an error in getting the full hashes, we need to treat the URL as safe.

Reply:
Also fixed.